Allure Gradle plugin 3.0.0: allureServe task always skipped (NO-SOURCE) with Kotlin DSL (build.gradle.kts)

When using the Allure Gradle plugin version 3.0.0 with a Kotlin DSL build file (build.gradle.kts), running `./gradlew allureServe` always returns:
> Task :allureServe NO-SOURCE
Skipping task ':allureServe' as it has no source files and no previous output files.

The task does not start the Allure server and does not generate any report.
This issue does not occur with plugin version 2.12.0 (everything works as expected).
Also, when switching to Groovy DSL (build.gradle), the allureServe task works correctly even with version 3.0.0.

**To Reproduce**
1. Use `build.gradle.kts` with Allure Gradle plugin version 3.0.0
2. Run tests to generate results
3. Run `./gradlew allureServe`
4. Observe that the task is skipped with NO-SOURCE


**Expected behavior**
The allureServe task should start the Allure server and display the report, as it does with Groovy DSL or with plugin version 2.12.0.

**Environment:**

- Gradle Kotlin DSL (build.gradle.kts)
- Allure Gradle plugin: 3.0.0
- Works with: Allure Gradle plugin 2.12.0 and/or Groovy DSL

**Additional context**
Switching to Groovy DSL or downgrading the plugin resolves the issue. The problem only appears with Kotlin DSL and plugin version 3.0.0.

The same issue after updating Gradle version to 9.1.0 and plugin version to 3.0.0:
> Task :allureReport NO-SOURCE
Skipping task '...' as it has no source files and no previous output files.

Downgrading to plugin version 2.12.0 does not help because it fails with exception (with updated Gradle version):
> Could not create task of type 'AllureReport'
 > Caused by: java.lang.ClassNotFoundException: org.gradle.api.plugins.internal.DefaultDecoratedConvention

--- a/allure-plugin/src/test/kotlin/io/qameta/allure/gradle/report/KotlinDslReportTest.kt
+++ b/allure-plugin/src/test/kotlin/io/qameta/allure/gradle/report/KotlinDslReportTest.kt
@@ -1,0 +1,69 @@
+package io.qameta.allure.gradle.report
+
+import org.assertj.core.api.Assertions.assertThat
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+import java.io.File
+
+@RunWith(Parameterized::class)
+class KotlinDslReportTest {
+    @Rule
+    @JvmField
+    val tempDir = TemporaryFolder()
+
+    @Parameterized.Parameter
+    lateinit var version: String
+
+    companion object {
+        @JvmStatic
+        @Parameterized.Parameters(name = "[{0}]")
+        fun versions() = listOf(
+            arrayOf("9.0.0"),
+            arrayOf("8.14.3"),
+            arrayOf("8.11.1"),
+        )
+    }
+
+    @Test
+    fun `allureReport should reuse results from a previous Kotlin DSL test run`() {
+        val projectDir = tempDir.newFolder("junit4-kotlin")
+        File("src/it/junit4-kotlin").copyRecursively(projectDir, overwrite = true)
+        projectDir.resolve("settings.gradle.kts").createNewFile()
+
+        val testResult = runner(projectDir).withArguments(commonArgs("test")).build()
+        assertThat(testResult.task(":test")?.outcome)
+            .`as`("test task outcome")
+            .isEqualTo(TaskOutcome.SUCCESS)
+        assertThat(projectDir.resolve("build/allure-results"))
+            .`as`("Allure results directory after test run")
+            .isNotEmptyDirectory()
+
+        val reportResult = runner(projectDir).withArguments(commonArgs("allureReport")).build()
+        assertThat(reportResult.task(":allureReport")?.outcome)
+            .`as`("allureReport should not become NO-SOURCE for Kotlin DSL projects")
+            .isEqualTo(TaskOutcome.SUCCESS)
+        assertThat(projectDir.resolve("build/reports/allure-report/allureReport"))
+            .`as`("Generated Allure report")
+            .isNotEmptyDirectory()
+    }
+
+    private fun runner(projectDir: File) = GradleRunner.create()
+        .withProjectDir(projectDir)
+        .withGradleVersion(version)
+        .withPluginClasspath()
+        .withTestKitDir(projectDir.resolve(".gradle-testkit"))
+        .forwardOutput()
+
+    private fun commonArgs(vararg tasks: String) = listOf(
+        "--stacktrace",
+        "--info",
+        "-Porg.gradle.daemon=false",
+        "--no-watch-fs",
+        *tasks
+    )
+}

